src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs

```diff
@@ -400,10 +273,10 @@ public static int Compare(string? strA, int indexA, string? strB, int indexB, in
                     return CompareInfo.Invariant.Compare(strA, indexA, lengthA, strB, indexB, lengthB, GetCaseCompareOfComparisonCulture(comparisonType));
 
                 case StringComparison.Ordinal:
-                    return CompareOrdinalHelper(strA, indexA, lengthA, strB, indexB, lengthB);
+                    return SpanHelpers.SequenceCompareTo(ref Unsafe.Add(ref strA.GetRawStringData(), indexA), lengthA, ref Unsafe.Add(ref strB.GetRawStringData(), indexB), lengthB);
```

Comment:
Why not keep the internal `CompareOrdinalHelper` method around? One benefit is that we can contain all of the unsafe logic in that method rather than sprinkling it around throughout these call sites.

Reply:
Is only 2 call sites? Also its a 6 parameter method so spills two arguments?